sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/security/TLSArtifactsPersister.java

```diff
@@ -0,0 +1,140 @@
+package com.mesosphere.sdk.offer.evaluate.security;
+
+import com.mesosphere.sdk.dcos.SecretsClient;
+import com.mesosphere.sdk.dcos.secrets.Secret;
+import com.mesosphere.sdk.dcos.secrets.SecretsException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Stores TLSArtifacts to a secret service.
+ */
+public class TLSArtifactsPersister {
+
+    private SecretsClient secretsClient;
+    private String serviceName;
+
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    public TLSArtifactsPersister(SecretsClient secretsClient, String serviceName) {
+        this.secretsClient = secretsClient;
+        this.serviceName = serviceName;
+    }
+
+    /**
+     * Store TLSArtifacts set to secret service.
+     *
+     * This method expects the namespace to be empty and no secret is existing. To clean up the namespace
+     * use {@link TLSArtifactsPersister ::cleanUpSecrets} call.
+     * @param tlsArtifacts
+     */
+    public void persist(
```

Comment:
Why does this persist _everything_ when the user may only have specified TLS and not KEYSTORE?